compiler/codegen/CCData.hpp

```diff
@@ -222,6 +239,22 @@ class CCData
       TR::Monitor      *_lock;
    };
 
+inline
+bool CCData::put(const void * const value, const size_t sizeBytes, const size_t alignmentBytes, const key_t * const key, index_t &index)
+   {
+   if (value == NULL)
+      {
+      return false;
+      }
+   return put_impl(value, sizeBytes, alignmentBytes, key, index);
+   }
+
+inline
+bool CCData::reserve(const size_t sizeBytes, const size_t alignmentBytes, const key_t * const key, index_t &index)
+   {
+   return put_impl(NULL, sizeBytes, alignmentBytes, key, index);
+   }
```

Comment:
My thinking was that it would be more convenient to have them in the header that you'd likely always include and leave all of the templated functions in the other header, but in hindsight anyone using `reserve()` will need to use `get<T>()` to retrieve a pointer so they'll end up including that header too, so no convenience gained at all.

Fixed.